### ISSUE TITLE
Possible error on train_svms.py

### ISSUE THREAD

**Issue by guyrose3** [rbgirshick/fast-rcnn#69]
after training a single svm, the trainer returns the weights w multiplied by the scale factor.
It's not clear to me why b is also multiplied by the scale factor.
In addition, I don't get the same score when applying these w,b to the net,
comparing to running the svm decision function with the scaled feature.
after removing the scale from b, I get identical results

**Comment by rbgirshick** [comment]
Thanks for spotting this issue; it is a bug. The bias shouldn't be scaled. This bug will lead to incorrect decision function values. However, it shouldn't change the class APs returned by the post-hoc SVM network. The scaled bias will just shift the scores of all detections by the same constant offset. That shift leaves the relative order of the detections unchanged. Nevertheless, this needs fixing...

**Comment by guyrose3** [comment]
I think another thing is the selection of hard negatives:
when the net bias is wrong, It won't produce the expected svm decision function values,
which in turn would cause to wrong selection of the hard negatives.

**Comment by rbgirshick** [comment]
Good point. Hopefully I'll have cycles to see how this effects the SVM experiments. Or maybe you already have some results?

**Comment by guyrose3** [comment]
unfortunately not yet.
I'm working with my own dataset, not Pascal VOC.

**Comment by rbgirshick** [comment]
I re-trained models from scratch and then tested softmax vs. svm after fixing the scaled-bias bug. Here are the new results. (The softmax results differ from those in the paper because these runs use a different random seed.)

mAP on VOC2007 test:

model | softmax | svm
--------|-------|-------
CaffeNet | 56.4 | 55.9
VGG M | 59.6 | 59.3
VGG 16 | 67.3 | 67.0

The feature_scale ranged from around .7 to .9 and so it was shrinking the negative biases towards 0 in the buggy experiments. This bias shrinkage had the effect of including more negative examples, not fewer, which is why the overall conclusion didn't change.

**Comment by rbgirshick** [comment]
Fixed in master.
